### PR TITLE
Fix over-estimated rate

### PR DESCRIPTION
**Pull request by kcwu**
The formula of average rate should be something similar to
`1/average(t for t in samples)`. But current implementation is similar to
`average(1/t for t in samples)`, which is incorrect.

Following code can demo this problem
```python
import random
import time
from tqdm import trange

for i in trange(10000, smoothing=0.01, leave=True, miniters=1, mininterval=0, ncols=0):
    time.sleep(random.random() / 100)
```

Output before this fix:
```
100% 10000/10000 [00:53<00:00, 413.96it/s]
```
After:
```
100% 10000/10000 [00:54<00:00, 187.06it/s]
```
10000/54=185.18 near 187.06

### CODE REVIEW

File: tqdm/_tqdm_gui.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ def __iter__(self):
 
                     ax.set_title(format_meter(
                         n, total, elapsed, 0,
-                        self.desc, ascii, unit, unit_scale, avg_rate,
+                        self.desc, ascii, unit, unit_scale, 1 / avg_time,
```
Comment (lrq3000) [inline tqdm/_tqdm_gui.py]: Should be `1 / avg_time if avg_time else None` like in _tqdm.py

File: tqdm/_tqdm_gui.py [hunk 2]
```diff
@@ -298,7 +298,7 @@ def update(self, n=1):
                 ax.set_title(format_meter(
                     self.n, total, elapsed, 0,
                     self.desc, self.ascii, self.unit, self.unit_scale,
-                    self.avg_rate, self.bar_format),
+                    1 / self.avg_time, self.bar_format),
```
Comment (lrq3000) [inline tqdm/_tqdm_gui.py]: Here too, should be `1 / self.avg_time if self.avg_time else None`.

**Comment by lrq3000** [comment]
Indeed, the previous computation was biased (but only when smoothing was enabled), but the primary goal was to facilitate dynamic adaptation for unregularly long iterations. Your PR should fix the issue, but we have to first check if the dynamic adaptation is OK (else we will need to fix that too :) ).

**Comment by kcwu** [comment]
Patch revised.

**Comment by lrq3000** [comment]
Wow, thank's @kcwu , but you didn't need to, I intended these notes more as reminders for tqdm devs :)

**Comment by kcwu** [comment]
FYI, slightly revised the patch and add my employer to the copyright line.

**Comment by casperdcl** [comment]
merged (and rebased) without the `delta_it != 0` assumption. `update(0)` will be interpreted as `update(1)` internally by default...

**Pull request closed without merging**